### ISSUE TITLE
FEEL engine fails to evaluate some valid expressions

### ISSUE THREAD

**Issue by rusterholz** [camunda/zeebe#8740]
**Describe the bug**

Some FEEL expressions which were valid in versions of Zeebe up through 1.3.0 are no longer processed correctly as of Zeebe 1.3.2.

**To Reproduce**

*It is not currently known exactly which expressions may cause this bug, or whether other conditions are necessary in order for it to be triggered.*

The specific expressions from our BPMN diagrams which were formerly working are output mappings on a Message Start Event:
* `= caterer_ever_responded or (last_message_initiated_by = "caterer_account")`
* `= customer_ever_responded or (last_message_initiated_by = "customer")`

On Zeebe 1.3.0, these expressions work as expected (they evaluate to true if either the first term is true or the parenthetical expression is true). **We have not tested on Zeebe 1.3.1,** but on Zeebe 1.3.2, these expressions fail to evaluate, and any running process instance receives an incident. The incident has type IO_MAPPING_ERROR and the error message:

> failed to evaluate expression '{message:if (message = null) then {id: message_id,initiator_type: last_message_initiated_by,status: message_status,content: content} else put all(message,{id: message_id,initiator_type: last_message_initiated_by,status: message_status,content: content}),caterer_ever_responded: caterer_ever_responded or (last_message_initiated_by = "caterer_account"),customer_ever_responded: customer_ever_responded or (last_message_initiated_by = "customer")}': no function found with name 'caterer_ever_responded or' and 1 parameters

We strongly suspect that this change may be due to the update of the feel-engine dependency in #8529 which was shipped in Zeebe 1.3.1.

**Expected Behavior**

1. Zeebe should correctly parse and process all valid FEEL expressions.
2. If a FEEL expression is invalid (or the FEEL engine cannot correctly parse and process it), the deployment of the process definition containing that expression should fail, so that the error can be caught before a process instance is created.
3. Unit and/or integration tests in the Zeebe codebase should verify that there have been no changes in how FEEL expressions are parsed and processed between versions of the feel-engine dependency (or that any changes present are expected by the Zeebe team). 
4. Dependency updates which contain breaking changes should not be included in patch version releases, but instead should result in minor or major version increments.

**Log/Stacktrace**

No stacktrace as this bug results in incidents in running instances, rather than in Java exceptions.

**Environment:**
- OS: Debian GNU/Linux 10 in a Docker container
- Zeebe Version: v1.3.2
- Configuration: zeebe-kafka-exporter v3.0.0

**Comment by saig0** [comment]
@rusterholz thank you for raising this up :+1: I'm sorry for the inconvenience. We dumped the FEEL version in `1.3.1` because it contains also some important bug fixes. We didn't expect that it results in invalid expressions. Unfortunately, we can't test every possible expression in Zeebe or the FEEL engine. But thanks to you and others, we can increase the coverage of expressions to avoid further regressions.  

I created an issue for the FEEL engine: https://github.com/camunda/feel-scala/issues/397

**Comment by saig0** [comment]
## Workaround

It should be possible to avoid this issue by removing the parentheses in the expressions.

Instead  of 

```
= caterer_ever_responded or (last_message_initiated_by = "caterer_account")
```

we can write

```
= caterer_ever_responded or last_message_initiated_by = "caterer_account"
```

**Comment by saig0** [comment]
This bug is fixed since version `8.0.0` :tada:
